doc/source/whatsnew/v1.0.0.rst

```diff
@@ -208,6 +208,7 @@ Other
 ^^^^^
 - Trying to set the ``display.precision``, ``display.max_rows`` or ``display.max_columns`` using :meth:`set_option` to anything but a ``None`` or a positive int will raise a ``ValueError`` (:issue:`23348`)
 - :meth:`DataFrame.to_csv` and :meth:`Series.to_csv` now support dicts as ``compression`` argument with key ``'method'`` being the compression method and others as additional compression options when the compression method is ``'zip'``. (:issue:`26023`)
+- Bug in :meth:`DataFrame.drop` when passing empty :meth:`DatetimeIndex` or `list`
```

Comment:
`list` should be double-backtics (or I suppose :class:\`list\` may work).

Can you add a :issue: reference like the others?

Reply:
Using you original example, it seems to not just be the list being empty.

```python
df.drop('a')
df.drop(['a'])
```

both raise with `TypeError: 'NoneType' object is not iterable`.

Is the condition that `labels` doesn't have any keys present in the DatetimeIndex?

Reply:
Good point, but in those cases `KeyError` would be more appropriate since `'a'`/`['a']` aren't in the index, wouldn't it?

Reply:
Ah, sorry, you're correct.

Reply:
`DatetimeIndex` should have `:class:`, not `:meth:`, and I think double-backticks for `list`

Reply:
My apologies. I'm not really familiar with rst. Will be fixing.